File: Laba2/24.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Введите числа через пробел:");
        String input = scanner.nextLine();
        String[] num = input.split(" "); // Разбиваем строку по пробелам
        for (int pos = 0; pos < num.length; pos++) {
            int numch = Integer.parseInt(num[pos]); // Переводим строку в число
            int n = Math.abs(numch), sum = 0, mult = 1;
            while (n > 0) {
                int digit = n % 10;
                sum += digit;
                mult *= digit;
                n /= 10;
            }
            if (sum < mult) {
                System.out.print(pos + " ");
            }
        }
    }
}

File: Laba2/24.kt
import java.util.Scanner

fun main() {
    val scanner = Scanner(System.`in`)
    println("Введите числа через пробел:")
    val input = scanner.nextLine()
    val num = input.split(" ").toTypedArray() // Разбиваем строку по пробелам
    for (pos in num.indices) {
        val numch = num[pos].toInt() // Переводим строку в число
        var n = Math.abs(numch)
        var sum = 0
        var mult = 1
        while (n > 0) {
            val digit = n % 10
            sum += digit
            mult *= digit
            n /= 10
        }
        if (sum < mult) {
            print("$pos ")
        }
    }
}
